src/core/ext/filters/client_channel/resolver/xds/xds_resolver.cc

```diff
@@ -731,13 +732,16 @@ XdsResolver::XdsConfigSelector::GetCallConfig(GetCallConfigArgs args) {
         method_config->GetMethodParsedConfigVector(grpc_empty_slice());
     call_config.service_config = std::move(method_config);
   }
-  call_config.call_attributes[XdsClusterAttributeTypeName()] = it->first;
+  ServiceConfigCallData::Pack(&call_config.call_attributes,
```

Comment:
It seems like the main benefit of `StringViewAttribute` is the `FromCallData()` method.  However, as per my comment elsewhere, I think that method is not going to be useful for LB policies, since they will need to use `ClientChannel::LoadBalancedCall::LbCallState::GetCallAttribute()` instead of `ServiceConfigCallData::GetCallAttribute()`.  Given that, I'm not sure it's actually that useful to provide this common class.

I don't think there's much overhead in defining a separate class for each string attribute, so it doesn't seem like something we should be working that hard to optimize away.  And having a separate subclass for each attribute provides a clean way to make it clear what the type name is for each attribute, because we can have a `static TypeName Type()` method, as per the example I showed above.  That more cleanly shows what data is associated with that typename, as opposed to the current approach of just exposing a stand-alone function to return a `UniqueTypeName` and then expecting the caller to magically know what type the associated value is.

Reply:
Done